### PR TITLE
Issue 44.mahchiahui.1

### PR DESCRIPTION
**Pull request by mahchiahui**
# Pull request questions

## Which issue does this address

Issue number: #44 

## Why was change needed

steps towards building a cd pipeline

## What does change improve

add build pipeline

### CODE REVIEW

File: .github/workflows/cd-pr-test-rhel-build.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Pull Request Test Senzing base Build on RHEL
+
+on:
+  pull_request:
+    branches:
+      - issue-44.mahchiahui.1
```
Comment (docktermj) [inline .github/workflows/cd-pr-test-rhel-build.yml]: This should be "master".  Correct?

Comment (mahchiahui) [reply .github/workflows/cd-pr-test-rhel-build.yml]: yes, i have made that change

File: .github/workflows/cd-3hr-cron-test-debian-build.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    steps:
+      - uses: actions/checkout@v1
+      - name: Print Environment Variables
+        run: printenv
+      - name: Download And Install Senzing Deb Package Onto Runner
+        id: download_install_senzing_deb
+        env:
+          SENZING_REPO_URL: https://senzing-staging-apt.s3.amazonaws.com/senzingstagingrepo_1.0.0-1_amd64.deb
+        run: |
+          wget ${SENZING_REPO_URL} >> output.log 2>&1
```
Comment (bryan-nice) [inline .github/workflows/cd-3hr-cron-test-debian-build.yml]: This is not installing files on the host. Please update so the files are installed on the host so the docker build process can use.

File: .github/workflows/cd-3hr-cron-test-debian-build.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          sudo apt-get -y install ./${SENZING_REPO_URL##*/} >> output.log 2>&1
+          rm -rf senzingrepo_1.0.0-1_amd64.deb >> output.log 2>&1
+          sudo apt-get update -y >> output.log 2>&1
+      - name: Docker build Senzing Base
+        id: build_senzing_base
+        run: |
+          docker build . >> build_failure.log 2>&1
```
Comment (bryan-nice) [inline .github/workflows/cd-3hr-cron-test-debian-build.yml]: This is not following the make file pattern. Please update to use the makefile commands to build the docker image.

File: .github/workflows/cd-pr-test-debian-build.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  prBuildSenzingBase:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Download And Install Senzing Deb Package Onto Runner
+        id: download_install_senzing_deb
+        env:
+          SENZING_REPO_URL: https://senzing-staging-apt.s3.amazonaws.com/senzingstagingrepo_1.0.0-1_amd64.deb
+        run: |
+          wget ${SENZING_REPO_URL} >> output.log 2>&1
```
Comment (bryan-nice) [inline .github/workflows/cd-pr-test-debian-build.yml]: This needs to follow the make file pattern. Use the make file to build the container. We are testing the behavior of make with the docker file. please fix.

File: .github/workflows/cd-pr-test-debian-build.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          sudo apt-get -y install ./${SENZING_REPO_URL##*/} >> output.log 2>&1
+          rm -rf senzingrepo_1.0.0-1_amd64.deb >> output.log 2>&1
+          sudo apt-get update -y >> output.log 2>&1
+      - name: Docker build Senzing Base
+        id: build_senzing_base
+        run: |
+          docker build . >> build_failure.log 2>&1
```
Comment (bryan-nice) [inline .github/workflows/cd-pr-test-debian-build.yml]: This is not following the make file pattern. Please update to use the makefile commands to build the docker image.

File: .github/workflows/cd-3hr-cron-test-rhel-build.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
+name: Cron Schedule 3 hr Test senzing base Build on RHEL
```
Comment (bryan-nice) [inline .github/workflows/cd-3hr-cron-test-rhel-build.yml]: This file is no longer needed

File: .github/workflows/cd-pr-test-rhel-build.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+name: Pull Request Test Senzing base Build on RHEL
```
Comment (bryan-nice) [inline .github/workflows/cd-pr-test-rhel-build.yml]: This file is no longer needed

File: .github/workflows/docker/ci-cd-debian-build/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,39 @@
+FROM senzing/base-image-debian:1.0.0
```
Comment (bryan-nice) [inline .github/workflows/docker/ci-cd-debian-build/Dockerfile]: This file is no longer needed

File: .github/workflows/docker/ci-cd-rhel-build/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,37 @@
+FROM senzing/base-image-centos:1.0.0
```
Comment (bryan-nice) [inline .github/workflows/docker/ci-cd-rhel-build/Dockerfile]: This file is no longer needed

**Pull request merged**